AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Models/SQLModelValueConverterTests.swift

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import Amplify
+@testable import AWSDataStoreCategoryPlugin
+
+class SQLModelValueConverterTests: XCTestCase {
+
+    let testDateString = "2020-02-02T08:00:00.000Z"
+
+    var testDate: Date {
+        return testDateString.iso8601Date!
+    }
+
+    func testModelWithEveryTypeConversionToBindings() {
+
+        // non-model type
+        let nonModelExample = ExampleNonModelType(someString: "string", someEnum: .foo)
+        let nonModelJSON = "{\"someString\":\"string\",\"someEnum\":\"foo\"}"
+
+        // example model
+        let example = ExampleWithEveryType(id: "df5dd4a4-34f8-4974-8a37-2617cf8dafe1",
```

Comment:
nice, this is a Model instance containing a `nonModelField` which is what i expected to be the use case. are we still supporting DataStore.save on NonModelTypes or it is always on a Model which may or may not contain non model fields?

Reply:
You can only save a `Model`, the non-model types can be used as values of a `Model`. That is also true on the AppSync level.